content/blog/package-react-components/index.md

```diff
@@ -0,0 +1,776 @@
+include your **Passively Compiled** dependency styles.
+
+Same goes for other assets such as fonts and images. Since Applications are treating
+dependencies as sources then all these things will be covered by whatever
+strategy the Application's Webpack configuration uses, like `file-loader`, etc.
+
+Providing a **Theming** API is also pretty simple as long as you have SASS or LESS
+core variables that define your theme and from which the rest of the library's
+styles depend on and as long as the **Component Consumers** support these type of preprocessing.
+
+##### Disadvantages
+
+You must either specify or follow a specification of what Webpack loaders,
+loader configuration, CSS configuration, Babel configuration, etc your library
+requires in order for Apps to compile it successfully. You could also construct
+some sort of webpack composition hook but that can get really complicated really
+soon without the proper care. An obvious problem with config composition is
+conflicts with a given loader configuration, such as babel, where the App uses
+certain configurations and the library uses others. In general I would say that
+the library and the consumer App cannot have too different bundling configurations
+if they want to work properly together, which means of course _tight coupling_.
+
+You have less encouragement to create a library public API and
+simply let _Component Consumers_ use any of your library files
+directly i.e. `import something from 'A/src/some/deep/nested/file.js'`.
+
+This will make even harder to migrated your library to an **Actively Compiled**
+strategy down the road because _Component Consumer_ Apps will be tightly coupled
+to your library's internal file structure and providing that same per-file API
+in Webpack is not trivial and has caveats (sharing code and such).
+
+By far, the **biggest disadvantage** with **Passive compilation** is that it will
+**infect** everything that it touches.
+If you are creating a new package that is **Actively compiled**
+but it uses some of the **Passively compiled** packages then BINGO,
+you need to customize your package build to compile those **passively compiled**
+packages and you end up in a very similar situation than simply making
+your new package **Passively compiled**. Which locks you down into this paradigm
+and all its flaws.
+
+I have been able to partially solve this situation by making those
+**Passively compiled** dependencies be `peerDependencies`, but you still need to
+handle them somehow in a test environment (probably by mocking them entirely,
+[Jest][17] does make this fairly easy)
+and if you have any sort of playground or documentation page where
+you display your component then BINGO again, you need to compile
+**Passively compiled** dependencies to make it work.
+
+If this seems crazy to you I can already tell you that I have
+experienced this in my day to day Job and is super frustrating.
```

Comment:
Great section as well. I think the previous one (Active vs Passive) is focused more about why not to use passive than comparing them objectively, but some of those disadvantages leak here.

Reply:
That was the intentional spirit, I want to be opinionated here and provide a barrier for people trying to do this (because it might be a super easy and viable solution for starters) and make them reconsider jajajaja.

I know this by heart (unfortunately)